tools/gfx/cuda/render-cuda.cpp

```diff
@@ -0,0 +1,1506 @@
+        {
+            entryPointHostData = alloca(entryPointDataSize);
+            cudaMemcpy(
+                entryPointHostData,
+                (void*)entryPointBuffer->getBindlessHandle(),
+                entryPointDataSize,
+                cudaMemcpyDeviceToHost);
+        }
+        //
+        // Second, the argument data for the entry-point parameters has
+        // been allocated and filled in as a single buffer, but `cuLaunchKernel`
+        // defaults to taking pointers to each of the kernel arguments.
+        //
+        // We could loop over the entry-point parameters using the refleciton
+        // information, and set up a pointer to each using the offset stored
+        // for it in the reflection data. Such an approach would require
+        // us to create and fill in a dynamically-sized array here.
+        //
+        // Instead, we take advantage of a documented but seldom-used feature
+        // of `cuLaunchKernel` that allows the argument data for all of the
+        // kernel "launch parameters" to be specified as a single buffer.
+        //
+        
+        void* extraOptions[] = {
+            CU_LAUNCH_PARAM_BUFFER_POINTER,
+            (void*)entryPointHostData,
+            CU_LAUNCH_PARAM_BUFFER_SIZE,
+            &entryPointDataSize,
+            CU_LAUNCH_PARAM_END,
+        };
+
+        // Once we have all the decessary data extracted and/or
+        // set up, we can launch the kernel and see what happens.
+        //
+        auto cudaLaunchResult = cuLaunchKernel(
+            currentPipeline->shaderProgram->cudaKernel,
+            x,
+            y,
+            z,
+            int(threadGroupSize[0]),
+            int(threadGroupSize[1]),
+            int(threadGroupSize[2]),
+            sharedSizeInBytes,
+            0,
+            nullptr,
+            extraOptions);
+
+        SLANG_ASSERT(cudaLaunchResult == CUDA_SUCCESS);
+    }
+
+    virtual void submitGpuWork() override {}
+
+    virtual void waitForGpu() override
+    {
+        auto result = cudaDeviceSynchronize();
+        SLANG_ASSERT(result == CUDA_SUCCESS);
+    }
+
+    virtual RendererType getRendererType() const override { return RendererType::CUDA; }
+
+public:
+    // Unused public interfaces. These functions are not supported on CUDA.
+    virtual const Slang::List<Slang::String>& getFeatures() override
+    {
+        static Slang::List<Slang::String> featureSet;
+        return featureSet;
+    }
+    virtual void setClearColor(const float color[4]) override
+    {
+        SLANG_UNUSED(color);
+    }
+    virtual void clearFrame() override {}
+    virtual void presentFrame() override {}
+    virtual TextureResource::Desc getSwapChainTextureDesc() override
+    {
+        return TextureResource::Desc();
+    }
+    
+    virtual Result createSamplerState(SamplerState::Desc const& desc, SamplerState** outSampler) override
+    {
+        SLANG_UNUSED(desc);
+        *outSampler = nullptr;
```

Comment:
Should the CUDA implementation allocate and return a kind of placeholder sampler object, with no actual functionality?